### PR TITLE
Add a superclass for literals

### PR DESCRIPTION
**Pull request by owen-mc**

### CODE REVIEW

File: ql/src/go.dbscheme [hunk 1]
```diff
@@ -223,6 +223,8 @@ case @expr.kind of
 
 @bitwiseexpr = @bitwiseunaryexpr | @bitwisebinaryexpr;
 
+@literal = @basiclit | @funclit | @compositelit;
```
Comment (sauyon) [inline ql/src/go.dbscheme:226]: You should make these changes in `extractor/dbscheme/tables.go` or they won't persist as the `make` script overwrites this file. Also, I think this change requires a (trivial) database upgrade script; I think there's a wiki page describing it on the old Semmle wiki, do you have access to that?

Comment (owen-mc) [reply ql/src/go.dbscheme:226]: No

Comment (owen-mc) [reply ql/src/go.dbscheme]: I've realised it's better to try to do it without changing the database scheme at all

File: ql/src/semmle/go/Expr.qll [hunk 2]
```diff
@@ -212,6 +212,23 @@ class Ellipsis extends @ellipsis, Expr {
...
+ *
+ * Examples:
+ *
+ * ```go
+ * "hello"
+ * func(x, y int) int { return x + y }
+ * map[string]int{"A": 1, "B": 2}
+ * ```
+ */
+class Literal extends @expr, Expr {
```
Comment (max-schaefer) [inline ql/src/semmle/go/Expr.qll:226]:
```suggestion
class Literal extends Expr {
```

**Pull request merged**